usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -2884,14 +3191,24 @@ arc_hdr_realloc(arc_buf_hdr_t *hdr, kmem_cache_t *old, kmem_cache_t *new)
 	ASSERT((old == hdr_full_cache && new == hdr_l2only_cache) ||
 	    (old == hdr_l2only_cache && new == hdr_full_cache));
 
+	/*
+	 * if the caller wanted a new full header and the header is to be
+	 * encrypted we will actually allocate the from the full crypt
```

Comment:
"we will actually allocate the from the full crypt cache"
Is a word missing? will allocate the <what>?